### PR TITLE
[ext-authz] add e2e test to ingress gateway

### PR DESCRIPTION
**Pull request by xulingqing**
[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[X] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[X] Does not have any changes that may affect Istio users.

### CODE REVIEW

File: tests/integration/security/authorization_test.go [hunk 1]
```diff
@@ -1363,5 +1368,36 @@ func TestAuthorization_Custom(t *testing.T) {
...
+					Name:     Name,
+					Host:     Host,
+					Path:     Path,
+					IP:       IP,
+					WantCode: WantCode,
+				}
+			}
+
+			ingressCases := []ingressCase{
+				newIngressCase("deny www.company.com", "www.company.com", "/", "172.16.0.1", 403),
```
Comment (yangminzhu) [inline tests/integration/security/authorization_test.go:1390]: I don't think you should use the `newIngressCase`, just modify and use the same `newTestCase` to add the tests for ingress gateway.

Also you don't need to use the host or IP, just send the requests at different paths like other tests should be good enough.

File: tests/integration/security/authorization_test.go [hunk 2]
```diff
@@ -1363,5 +1368,36 @@ func TestAuthorization_Custom(t *testing.T) {
...
+				newIngressCase("allow www.company.com", "www.company.com", "/custom", "172.16.0.1", 200),
+			}
+			for _, tc := range ingressCases {
+				ctx.NewSubTest(tc.Name).Run(func(ctx framework.TestContext) {
+					headers := map[string][]string{
+						"X-Forwarded-For": {tc.IP},
```
Comment (yangminzhu) [inline tests/integration/security/authorization_test.go:1396]: No need to set the `X-Forwarded-For`, also set `x-ext-authz` like other test cases (some use allow and some use deny).

File: tests/integration/security/authorization_test.go [hunk 3]
```diff
@@ -1301,9 +1309,8 @@ func TestAuthorization_Custom(t *testing.T) {
 			if _, _, err := kube.WaitUntilServiceEndpointsAreReady(ctx.Clusters().Default(), extAuthzServiceNamespace.Name(), "ext-authz"); err != nil {
 				ctx.Fatalf("Wait for ext-authz server failed: %v", err)
 			}
-
-			policy := applyYAML("testdata/authz/v1beta1-custom.yaml.tmpl", ns)
-			defer ctx.Config().DeleteYAMLOrFail(t, ns.Name(), policy...)
+			policy := applyYAML("testdata/authz/v1beta1-custom.yaml.tmpl", nil)
```
Comment (yangminzhu) [inline tests/integration/security/authorization_test.go:1312]: I'm not sure if this is going to work, should the policy for istio-system namespace be put in a separate yaml?

Comment (xulingqing) [reply tests/integration/security/authorization_test.go:1312]: I seperate the ingress part in this pr https://github.com/istio/istio/pull/29899, but seems still got the timeout issue.

**Review by yangminzhu — COMMENTED** [review]
Looks good! Just a few small comments.

File: tests/integration/security/authorization_test.go [hunk 4]
```diff
@@ -1389,6 +1397,31 @@ func TestAuthorization_Custom(t *testing.T) {
 				newTestCase(x, f, "", "tcp-8093", "", true, scheme.TCP),
 			}
 
+			ingr := ist.IngressFor(ctx.Clusters().Default())
```
[the remark this answers is not in the document]
Comment (xulingqing) [reply tests/integration/security/authorization_test.go:1400]: Sure. Thanks!

**Review by yangminzhu — APPROVED** [review]

**Pull request merged**